src/marqo/core/exceptions.py

```diff
@@ -112,3 +112,15 @@ class DuplicateDocumentError(AddDocumentsError):
 
 class TooManyFieldsError(MarqoError):
     pass
+
+
+class DeviceError(MarqoError):
```

Comment:
How is this handled by the API layer?

Reply:
See https://github.com/marqo-ai/marqo/pull/1056/files#diff-ba0027ca86b6804c6a459ce5153a53376e7c99eae82e93bcde231c854b0d402aR115.